### DISCUSSION TITLE
Terminal Interface

### DISCUSSION THREAD

**Discussion by kilicbaran** [jotaen/xit#12] [Tooling]
This is not about specification but about tooling. Can we get some terminal love? There are probably a lot of tools for other todo formats. This is just a very simple bash script that can be added to `.bash_profile`. It can be extended for other options or a more proper version can be implemented in a programming language.

```bash
XIT_HOME=~/home.xit

xit() {

  if [ ! -f "$XIT_HOME" ]; then
    touch $XIT_HOME
  fi

  local subcommand=$1;
  shift;

  case $subcommand in
    open)
      if [[ $OSTYPE == 'darwin'* ]]; then
        command open -t $XIT_HOME
      else
        command nano $XIT_HOME
      fi
      
      ;;

    add|a) # Requires use of single quotes for the use of exclamation mark
      command echo "$*" >> $XIT_HOME
      ;;

    search|s) # Doesn't handle multiline
      command grep "$*" $XIT_HOME
      ;;

    * )
      command echo "Usage: ${FUNCNAME[0]} {open|add|search}"
      ;;
  esac
}
```

### Some example usage
- `xit open` will open the file.
- `xit add '[ ] some todo'` will add it to file.
- `xit search todo` will return the line containing the given word.

**Comment by jotaen** [comment]
👍 I think that kind of tooling would be terrific. I’m happy to link such tools to from [website](https://github.com/jotaen/xit).

**Comment by kevin-gwyrdh** [comment]
Hi 
Thanks for suggesting this @kilicbaran and thanks @jotaen for the whole [x]it project
I have taken your starting code and made a bash script to show tasks as a list and as a simple kanban. Two versions one trying to follow [x]it format and the other having some compatability with the todo.txt format. Code will be a bit clunky but seems to work. 
Code is here https://github.com/kevin-gwyrdh/xitbash

**Comment by hoechstleistungshaartrockner** [comment]
Hi everyone! I created a python based command line interface
https://github.com/hoechstleistungshaartrockner/xitkit

**Reply by jotaen** [reply]
Wow, this looks pretty full-featured! Also great to see comprehensive test coverage.

Let me know if you’d like me to link this from the “Tooling” section on https://xit.jotaen.net!

**Reply by hoechstleistungshaartrockner** [reply]
Sure, that'll be great, thanks!
